src/Domain/EmbeddingMaps/AffineMap.hpp

```diff
@@ -14,19 +14,27 @@
 
 namespace EmbeddingMaps {
```

Comment:
I agree with Will that since the embedding maps will only be used via an abstract base class there is nothing to be gained from having functions inline. It's probably best then that we move all the member functions, including constructor and migration constructor to having the definitions in cpp files for all the maps in this PR,sorry. Even if the function is one line, it cannot be inlined through calls to virtual base class pointers and so would only add to compilation time, as Will pointed out.